### PR TITLE
Add TRegex to the list of Truffle language implementations

### PR DESCRIPTION
**Pull request by Godin**

### CODE REVIEW

**Comment by nightscape** [comment]
For added coolness, the name should be spelled TRegEX 😄

**Comment by Godin** [comment]
@nightscape indeed **TR**eg**EX** looks more cooler :thumbsup:
but different from spelling at https://github.com/oracle/graal/tree/master/regex

**Comment by wirthi** [comment]
Hi @Godin 

Thanks for your PR. I am not sure whether we want to do this as TRegex is not a "public" language, but a mere DSL to be used by other languages. We would at least need to clarify that here, by adding it to a list of "internal" languages.

-- Christian

File: truffle/docs/Languages.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ This page is intended to keep track of the growing number of Truffle language im
 * [TruffleSOM](https://github.com/SOM-st/TruffleSOM), a SOM Smalltalk implementation. 
 * [SOMns](https://github.com/smarr/SOMns), a Newspeak implementation for Concurrency Research.
 * [SimpleLanguage](https://github.com/graalvm/simplelanguage), a toy language implementation to demonstrate Truffle features. 
+* [TRegex](https://github.com/oracle/graal/tree/master/regex), an implementation of a subset of ECMAScript regular expressions.
```
Comment (Godin) [inline truffle/docs/Languages.md]: @wirthi what do you think about

```suggestion
* [TRegex](https://github.com/oracle/graal/tree/master/regex), an internal available for use by other languages implementation of a subset of ECMAScript regular expressions.
```

?

Comment (boris-spas) [reply truffle/docs/Languages.md]: maybe: "An implementation of a subset of ECMAScript regular expressions (internal i.e. only for use by other languages.)

**Comment by boris-spas** [comment]
Is there still interest in merging this PR?
If the conflicts are resolved I'd try to shepherd it.

**Comment by Godin** [comment]
@boris-spas I resolved conflict and rebased this PR on top of master.
Also applied a slightly modified version of [your suggestion](https://github.com/oracle/graal/pull/840#discussion_r368558493).

**Comment by boris-spas** [comment]
Cool! Thanks!

**Pull request merged**